### PR TITLE
Reduce test

### PR DESCRIPTION
**Pull request by CongMa13**
Added unit tests of reduction:
* rank2-6

Found that some test cases could pass:
* when beta != 0
* when reducing the fastest growing dim
* when datatype is bf16 or f16

Comment out these test cases. Waiting for CK team to fix them.

### CODE REVIEW

File: library/src/reduction/reduction_solution_bf16_bf16_instance.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ *******************************************************************************/
+
+#include "hiptensor/internal/types.hpp"
+#include "reduction_solution.hpp"
+#include "reduction_solution_instances.hpp"
+
+namespace hiptensor
+{
+    void ReductionSolutionInstances::genReductionSolutionF32F32Instances()
```
Comment (cgmillette) [inline library/src/reduction/reduction_solution_bf16_bf16_instance.cpp]: Naming of this should be bf16?

File: library/src/reduction/reduction_solution_f16_f16_instance.cpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ *
+ *******************************************************************************/
+
+#include "hiptensor/internal/types.hpp"
+#include "reduction_solution.hpp"
+#include "reduction_solution_instances.hpp"
+
+namespace hiptensor
+{
+    void ReductionSolutionInstances::genReductionSolutionF32F32Instances()
```
Comment (cgmillette) [inline library/src/reduction/reduction_solution_f16_f16_instance.cpp]: Naming of this should be f16?

Comment (cgmillette) [reply library/src/reduction/reduction_solution_f16_f16_instance.cpp]: Also this is the 4_2 case only. Are we waiting for CK to update the M6 kernel?

File: test/03_reduction/reduction_test.cpp [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        EXPECT_TRUE((lengths.size() > 1) && (lengths.size() <= 6));
+        EXPECT_TRUE((outputDims.size() >= 1) && (outputDims.size() <= 6));
+
+        EXPECT_TRUE((op == HIPTENSOR_OP_ADD) || (op == HIPTENSOR_OP_MUL) || (op == HIPTENSOR_OP_MAX)
+                    || (op == HIPTENSOR_OP_MIN));
+
+        EXPECT_EQ(testType.size(), 2); // HIP_R_16F or HIP_R_32F
+        auto acDataType      = testType[0];
+        auto computeDataType = convertToComputeType(testType[1]);
+        EXPECT_TRUE(
```
Comment (cgmillette) [inline test/03_reduction/reduction_test.cpp:106]: Reduction instances only include f32 and f64 for now. Shall we disable the F16 and BF16 tests?

**Review by cgmillette — COMMENTED** [review]
In general, looks great. Just a few comments from above. Please look into the docs build failures before approval!

**Comment by cgmillette** [comment]
This looks like it has changes from #230 as well?

**Review by cgmillette — CHANGES_REQUESTED** [review]
Please check other comments and have a look at broken docs build

File: library/src/reduction/reduction_solution_bf16_bf16_instance.cpp (hunk 1 above)
Comment (CongMa13) [reply library/src/reduction/reduction_solution_bf16_bf16_instance.cpp]: Right now, CK has not fully supported bf16 and f16. There are building error to build these instances. So this file is not in the CMakeLists.txt file. I will remove this file.

File: library/src/reduction/reduction_solution_f16_f16_instance.cpp (hunk 2 above)
Comment (CongMa13) [reply library/src/reduction/reduction_solution_f16_f16_instance.cpp]: Right now, CK has not fully supported bf16 and f16. There are building error to build these instances. So this file is not in the CMakeLists.txt file. I will remove this file.

File: test/03_reduction/reduction_test.cpp (hunk 3 above)
Comment (CongMa13) [reply test/03_reduction/reduction_test.cpp:106]: Right now, CK has not fully supported bf16 and f16. There are building error to build these instances. So we cannot test f16 and bf16 now.

**Comment by CongMa13** [comment]
> This looks like it has changes from #230 as well?

yeah, this includes #230 . I will rebase this PR after #230 is merged.

**Review by cgmillette — APPROVED** [review]
Looks good

**Comment by CongMa13** [comment]
> Please check other comments and have a look at broken docs build

docs have been built successfully

**Pull request merged**